src/index.ts

```diff
@@ -57,15 +59,86 @@ if (require.main === module) {
         forceZeroHop,
         rpcProvider,
     };
-    const settings = {
+    const settings: Settings = {
         pushGateway,
         intervalMs,
         offsetMs,
+        metrics: {},
+        metricLabels: {
+            hops: forceZeroHop ? '0' : '1',
+        },
     };
     const logOpts = {
         uHTTPsettings,
         settings,
     };
+
+    if (process.env.UHTTP_LM_METRIC_INSTANCE !== undefined) {
+        settings.metricLabels['instance'] = process.env.UHTTP_LM_METRIC_INSTANCE;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_REGION !== undefined) {
+        settings.metricLabels['region'] = process.env.UHTTP_LM_METRIC_REGION;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_ZONE !== undefined) {
+        settings.metricLabels['zone'] = process.env.UHTTP_LM_METRIC_ZONE;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_LOCATION !== undefined) {
+        settings.metricLabels['location'] = process.env.UHTTP_LM_METRIC_LOCATION;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_LATITUDE !== undefined) {
+        settings.metricLabels['latitude'] = process.env.UHTTP_LM_METRIC_LATITUDE;
+    }
+
+    if (process.env.UHTTP_LM_METRIC_LONGITUDE !== undefined) {
+        settings.metricLabels['longitude'] = process.env.UHTTP_LM_METRIC_LONGITUDE;
+    }
```

Comment:
I would prefer to move those to line 55, so all the env var handling is in one place